### ISSUE TITLE
Order of `Template` fields doesn't match display order

### ISSUE THREAD

**Issue by halleysfifthinc** [invenia/PkgTemplates.jl#286]
I just noticed that the order of the fields in `Template` doesn't match the order they are shown in:

```julia
Template:
  authors: ["John Smith <fakemail@example.com> and contributors"]
  dir: "~/.julia/dev"
  host: "github.com"
  julia: v"1.0.0"
  user: "johnsmith"
  plugins:
    ...
```
Where the `user` and `plugins` fields are swapped. I'm guessing that this is to show plugins at the end (given its non-constant length), but it tripped me up when I tried to use the constructor and assumed the order of display matched the arg order in the constructor.

Should the field order be updated to match the display order?

**Comment by nickrobinson251** [comment]
Thanks for the issue! I think the only constructor that's documented and considered public API is the all keywords constructor (where order doesn't matter) e.g.
 ```julia
tpl = Template(; user="halleysfifthinc", host="gitlab.com", juia=v"1.5")
```
or to construct a `Template` interactively with `tpl = Template(interactive=true)` (or a mix of these, e.g. `tpl = Template(; interactive=true, user="halleysfifthinc")`

I lean towards thinking that how things a Template is pretty-printed and it is constructed should be kept separate, so as to give us freedom about how to make Templates display nicely in the repl, without having the change the type itself. As you say, the plugins being displayed last makes everything a bit easier to see.

That said, in this case, to match the current printing it'd just be a matter of changing one line (and updating any internal code that uses the position-args constructor)
```diff
--- a/src/template.jl
+++ b/src/template.jl
@@ -74,8 +74,8 @@ struct Template
     dir::String
     host::String
     julia::VersionNumber
-    plugins::Vector{<:Plugin}
     user::String
+    plugins::Vector{<:Plugin}
 end
```
Still, i think we'd rather just keep recommending the keyword-only constructors.

Do you have a use-case for the non-keyword constructor (where order does matter)? Or how did you come across it?

**Comment by halleysfifthinc** [comment]
That's fair. You are right that it would be a simple change, albeit a breaking one.

I stumbled on this after I created a Template and then needed to change the `dir`, but couldn't since `Template` is immutable. I ended up running `Template(;dir="/new/dir", tpl.authors, tpl.host, tpl.julia, tpl.user, tpl.plugins)`. An easier solution to the problem I had would be to add a `Template` constructor that takes an existing template as a basis and uses any other specified keywords to override what is present in the given template. However, I don't know if my situation is common enough that such a constructor would be worth adding.
